apps/aecore/src/aec_dry_run.erl

```diff
@@ -64,8 +78,65 @@ add_account(#{pub_key := PK, amount := A}, AccountsTree) ->
         end,
     aec_accounts_trees:enter(Account, AccountsTree).
 
-dummy_sign_txs(Txs) ->
-    [ aetx_sign:new(Tx, [dummy_sign()]) || Tx <- Txs ].
+prepare_txs([]) -> [];
+prepare_txs([{tx, Tx} | Txs]) ->
+    [{tx, [], dummy_sign(Tx)} | prepare_txs(Txs)];
+prepare_txs([{call_req, Req} | Txs]) ->
+    [prepare_call_req(Req) | prepare_txs(Txs)].
+
+dummy_sign(Tx) ->
+    aetx_sign:new(Tx, [<<0:(?BLOCK_SIGNATURE_BYTES*8)>>]).
+
+prepare_call_req(ReqMap) ->
+    try %% Required
+        {ok, CallData} = aeser_api_encoder:safe_decode(contract_bytearray, maps:get(<<"calldata">>, ReqMap)),
+        {ok, CtPub}    = aeser_api_encoder:safe_decode(contract_pubkey, maps:get(<<"contract">>, ReqMap)),
+
+        %% Optional
+        Amount = maps:get(<<"amount">>, ReqMap, 0),
+        Caller = case maps:get(<<"caller">>, ReqMap, undefined) of
+                     undefined -> ?MR_MAGIC;
+                     EncCaller ->
+                         {ok, CallerX} = aeser_api_encoder:safe_decode(account_pubkey, EncCaller),
+                         CallerX
+                 end,
+        Gas    = maps:get(<<"gas">>, ReqMap, 1000000),
+        ABI    = maps:get(<<"abi_version">>, ReqMap, ?ABI_AEVM_SOPHIA_1),
+        Nonce  = maps:get(<<"nonce">>, ReqMap, 1),
+
+        {ok, CallTx} = aect_call_tx:new(#{caller_id   => aeser_id:create(account, Caller),
+                                          nonce       => Nonce,
+                                          contract_id => aeser_id:create(contract, CtPub),
+                                          abi_version => ABI,
+                                          fee         => 1000000 * 1000000,
+                                          amount      => Amount,
+                                          gas         => Gas,
+                                          gas_price   => 1000000,
+                                          call_data   => CallData}),
+
+        %% Other options
+        ContextMap = maps:get(<<"context">>, ReqMap, #{}),
+        TxHashCtxt = case maps:get(<<"tx_hash">>, ContextMap, none) of
+                         none -> [];
+                         TxHashEnc ->
+                            {ok, TxHash} = aeser_api_encoder:safe_decode(tx_hash, TxHashEnc),
+                            [{auth_tx, TxHash}]
+                     end,
+        StateCtxt  = [ stateless || maps:get(<<"stateful">>, ContextMap, false) /= true ],
+        Context    = TxHashCtxt ++ StateCtxt,
+        {tx, Context, dummy_sign(CallTx)}
+    catch _:_R ->
+        error({bad_dry_run_call_request, ReqMap})
+    end.
+
+prepare_env(Env, Opts) ->
+    case proplists:get_value(auth_tx, Opts, undefined) of
+        undefined -> Env;
+        TxHash    -> aetx_env:set_ga_tx_hash(Env, TxHash)
```

Comment:
Would it add value to check that TxHash is a binary hash?

Reply:
It comes from `aeser_api_encoder:safe_decode/2` so seems like overkill